### PR TITLE
Feature/2644/services grab bag2

### PR DESCRIPTION
**Pull request by garyluu**
Another grab bag of UI changes from the doc (not the issue)

The last three commits before "fix test" are the largest and are mainly refactoring, testing, and loading all action buttons at once.  Tool and workflow actions extends entry actions.

Also had to change some Cypress tests because it was still flakey

Successful runs 8/10 (reset DB, publish button)

I am constantly re-running the tests to find new flakey tests (that are not reset DB related).  You can review now.

### CODE REVIEW

File: src/app/workflow/workflow.component.html [hunk 1]
```diff
@@ -372,7 +334,7 @@ <h3>Recent Versions</h3>
               </p>
               <hr class="m-0" />
             </span>
-            <a (click)="selectTab(2)">See all versions</a>
+            <a (click)="selectVersionsTab()">See all versions</a>
```
Comment (coverbeck) [inline src/app/workflow/workflow.component.html]: 👍

File: src/app/shared/entry-actions/entry-actions.service.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  isEntryHosted(entry: DockstoreTool | Workflow | null): boolean {
+    if (DockstoreTool.ModeEnum.HOSTED !== Workflow.ModeEnum.HOSTED) {
+      console.error('ModeEnum.Hosted for all entries should be the same');
+    }
+    if (entry) {
+      return entry.mode === DockstoreTool.ModeEnum.HOSTED;
+    } else {
+      return true;
```
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/shared/entry-actions/entry-actions.service.ts]: I don't know, I just moved it here without changing it

File: src/app/shared/entry-actions/entry-actions.service.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  private isEntryValid(entry: Entry | null): boolean {
+    if (!entry) {
+      return false;
+    }
+    if (entry.is_published) {
+      return true;
+    }
+    const versionTags = entry.workflowVersions;
+
+    if (!versionTags || versionTags.length === 0) {
```
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/shared/entry-actions/entry-actions.service.ts]: hmm...might as well

File: cypress/integration/testThatDoNotModifyDatabase/services.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
-import { assertNoTab, getTab, goToTab, setTokenUserViewPort } from '../../support/commands';
```
Comment (denis-yuen) [inline cypress/integration/testThatDoNotModifyDatabase/services.ts]: Can't comment on the filename directly, but it sounds kinda yoda-like

File: src/app/shared/descriptor-type-compat.service.ts [hunk 5]
```diff
@@ -114,8 +114,7 @@ export class DescriptorTypeCompatService {
       case ToolDescriptor.TypeEnum.NFL:
         return 'PLAIN-NFL';
       case ToolDescriptor.TypeEnum.SERVICE:
-        // TODO: Actually handle this
-        return null;
+        return 'PLAIN-SERVICE';
```
Comment (denis-yuen) [inline src/app/shared/descriptor-type-compat.service.ts]: I rue the day I thought of the `plain` idea

File: src/app/shared/state/token.service.ts [hunk 6]
```diff
@@ -68,15 +68,19 @@ export class TokenService {
...
+  getGitHubOrganizations(token: string | null) {
+    if (token) {
+      const httpClient = new HttpClient(this.httpBackend);
+      const httpOptions = {
+        headers: new HttpHeaders({
+          'Content-Type': 'application/json',
+          Authorization: 'token ' + token
+        })
+      };
+      const getOrganizationUrl = 'https://api.github.com/user/orgs';
```
Comment (denis-yuen) [inline src/app/shared/state/token.service.ts]: Not a fan about this occurring in the UI as opposed to in the web-service, it seems a bit too one-of-a-kind.

Comment (garyluu) [reply src/app/shared/state/token.service.ts]: This PR just checks the null-checks the token.  Can put this in the webservice some other time.

File: src/app/shared/descriptor-type-compat.service.ts (hunk 5 above)
Comment (garyluu) [reply src/app/shared/descriptor-type-compat.service.ts]: Will there ever be a good time to propose a change for it?  or has that ship sailed

**Pull request merged**